jackson-datatype-problem/src/main/java/org/zalando/problem/ProblemModule.java

```diff
@@ -53,21 +55,21 @@ public ProblemModule() {
     @SafeVarargs
     public <E extends Enum & StatusType> ProblemModule(final Class<? extends E>... types)
             throws IllegalArgumentException {
-     
+
         this(false, buildIndex(types));
     }
 
-    public ProblemModule(boolean stacktraces, ImmutableMap<Integer, StatusType> statuses) {
+    public ProblemModule(boolean stacktraces, Map<Integer, StatusType> statuses) {
         this.stacktraces = stacktraces;
-        this.statuses = statuses;
+        this.statuses = Collections.unmodifiableMap(statuses);
```

Comment:
Do we need this constructor to be public at all? I can imagine someone providing wrong mapping